src/main/java/trade/wayruha/mexc/ws/AutoRenewalPrivateWSClient.java

```diff
@@ -39,16 +37,23 @@ protected void connect(Set<String> channels) {
             this.wsService = new PrivateWSSubscriptionService(apiClient);
         }
         final String listenKey = wsService.createListenKey();
+        log.trace("{} Got new private key: {} for WS channels: {}", logPrefix, listenKey, channels);
         this.connectionRequest = new Request.Builder()
                 .url(config.getWebSocketHost() + "?" + LISTEN_KEY_QUERY_PARAM + "=" + listenKey)
                 .build();
         super.connect(channels);
 
-        final KeepAliveTask keepAliveTask = new KeepAliveTask(listenKey);
-        this.scheduledKeepAliveTask = this.getScheduler().scheduleAtFixedRate(keepAliveTask,
+        this.scheduledKeepAliveTask = this.getPrivateScheduler().scheduleAtFixedRate(new KeepAlivePrivateWsTask(listenKey),
```

Comment:
Unfortunately - yes, we need 3 methods.

PingTask - keeps subscriptions to streams with no messages

KeepAliveTask -  keep listenKey (= access_token by meaning) to all kind of WS methods. If it goes bad, then new connections to the WS will not be possible and the old ones will have to fall off.

ResubscribeTask - reconnects all subscriptions to all method every 23 hours because it was said in MEXC docs, that WS subscription is possible only for 24 hours long and then it will block you. I haven’t tested a real case of blocking after 24 hours, and I don’t really want to :) Current implementation works and I'm happy.